component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java

```diff
@@ -70,4 +71,35 @@ public ComponentException(final Throwable cause) {
         this(ErrorOrigin.UNKNOWN, cause.getMessage(), cause);
     }
 
+    /**
+     * Convert all exception stack to generic throwable stack to avoid unknown exception at deserialization time..
+     * 
+     * @param t
+     * @return An Throwable.
+     */
+    protected Throwable toGenericThrowable(final Throwable t) {
```

Comment:
this method looks like InvocationExceptionWrapper.mapException, no? ;-)

Reply:
InvocationExceptionWrapper.mapException mainly for : toRuntimeException, not similar

Reply:
InvocationExceptionWrapper.mapException mainly for : toRuntimeException, not similar